python-sdk/src/astro/databases/snowflake.py

```diff
@@ -638,6 +639,22 @@ def _copy_into_table_from_stage(self, source_file, target_table, stage):
             self.drop_stage(stage)
         return rows
 
+    def _validate_before_copy_into(self, source_file, target_table, stage):
+        """Validate COPY INTO command to tests the files for errors but does not load them."""
+        if self.load_options.validation_mode is None:
+            return
+        table_name = self.get_table_qualified_name(target_table)
+        file_path = os.path.basename(source_file.path) or ""
+        sql_statement = (
+            f"COPY INTO {table_name} FROM "
+            f"@{stage.qualified_name}/{file_path} VALIDATION_MODE='{self.load_options.validation_mode}'"
+        )
+        try:
```

Comment:
@sunank200, how are we handling timeout for this operation? I remember we saw there are scenarios when it is very slow - it would be great if we handled this gracefully.

Reply:
@tatiana so validation mode actually doesn't load the data to the table it just validates all the data and files and throws the relevant errors. It instructs the COPY command to validate the data files instead of loading them into the specified table; i.e. the COPY command tests the files for errors but does not load them. 

There are three modes possible: `RETURN_n_ROWS` | `RETURN_ERRORS` | `RETURN_ALL_ERRORS`
- `RETURN_15_ROWS` for example validates the specified 15 rows if no errors are encountered; otherwise, fails at the first error encountered in the rows.
- `RETURN_ERRORS` returns all errors (parsing, conversion, etc.) across all files specified in the COPY statement.
- `RETURN_ALL_ERRORS` returns all errors across all files specified in the COPY statement, including files with errors that were partially loaded during an earlier load because the ON_ERROR copy option was set to CONTINUE during the load.

I think even for large files if users want to perform validations using this mode, it would just take double the time - the first half to validate and another half to just load the file. If the user doesn't wish to validate, in that case, it would throw the correct error anyways. That was my understanding after multiple runs for various queries I tried yesterday. WDYT, do we still require a timeout? I felt its not necessary in this case.